### PR TITLE
Package npm

### PR DESCRIPTION
**Pull request by nshCore**
bootstrap NPM packaging of wasm

### CODE REVIEW

**Comment by nshCore** [comment]
> Added some things. Should probably be linted before being merged.

yeh just trying to get the webpack system stable and working, should only need to lint the tests to airbnb-base standard auto-gen code shouldn't be linted

**Comment by nshCore** [comment]
I have a feeling the way this is configured it will only support nodejs modules, trying to enable web/ browser wasm-pack target throws never-ending babel issues, I suggest we merge this now, and open an issue and open a ticket @rustwasm team to help out

File: README.md [hunk 1]
```diff
@@ -24,13 +24,29 @@ If you currently have the stable release do
...
+nvm install lts/carbon
+nvm use lts/carbon
+```
+
+For further instructions on NVM refer to their repo https://github.com/nvm-sh/nvm
+
 You can build with:
 
-```rustup run nightly wasm-pack build --release --target web```
+```npm run build:prod```
```
Comment (franssl) [inline README.md]: I had to do:
npm install
npm run build (not build:prod)

Comment (nshCore) [reply README.md]: Hi @franssl sorry must be a mistake in the readme correct build is cmd is

```
npm run build
```

under the hood it invokes webpack build

https://github.com/catalyst-network/wasm-ed25519ph/blob/package-npm/build/webpack.common.babel.js#L45

running wasm-pack with the following flag

```
--target nodejs
```

This is the only build target I could get to build a bin that would run in the test environment.

when trying the web/browser/bundler targets it produces ES6 javascript and no matter what babel transpiler settings I set it wouldn't transpile the ES6 to vanilla js to execute the tests in jest. 

I imagine this error

```
Uncaught TypeError: TextDecoder is not a constructor
```

you see is when you run npm run build then try use the bundle in the browser?

Running npm run build && npm run test, which should do a node target and run as a node process passes the tests. 

Unfortunatly webpack wasm is not well covered online.

**Pull request merged**